### ISSUE TITLE
ZLS doesn't launch sometimes

### ISSUE THREAD

**Issue by BezBartek** [FalsePattern/ZigBrains#56]
**Pycharm 2024.1.4**
**zls: 0.13**
**zig 0:13**
Configuration:
![image](https://github.com/user-attachments/assets/d4c32682-ddfb-4bda-bb06-00d6d67a9fc8)
ZLS not work:
![image](https://github.com/user-attachments/assets/21181f7c-677f-4c9f-816a-fed37f8f4884)

Am I doing something wrong?
Maybe there is a lack of information in docs.

**Comment by Zelinzky** [comment]
Chech your plugin in File | Settings | Plugins. go into ZigBrains and take a look in aditional info. Version 16.0.0-241 has a dependency in another plugin called LSP4IJ from Red-Hat. Maybe you're missing that.

**Comment by BezBartek** [comment]
Hello @Zelinzky 
I have LSP4IJ installed:
![image](https://github.com/user-attachments/assets/3ce71f5d-9fb3-401d-9534-c24326da593a)

Logs from PyCharm, I can't "print" them all 
```
Caused by: java.lang.NullPointerException
	at java.base/java.util.Objects.requireNonNull(Objects.java:209)
	at java.base/sun.nio.fs.UnixFileSystem.getPath(UnixFileSystem.java:263)
	at java.base/java.nio.file.Path.of(Path.java:147)
	at com.falsepattern.zigbrains.project.toolchain.ToolchainZLSConfigProvider.lambda$getEnvironment$0(ToolchainZLSConfigProvider.java:38)
	at java.base/java.util.Optional.ifPresent(Optional.java:178)
	at com.falsepattern.zigbrains.project.toolchain.ToolchainZLSConfigProvider.getEnvironment(ToolchainZLSConfigProvider.java:38)
	at com.falsepattern.zigbrains.zig.environment.ZLSConfigProvider.findEnvironment(ZLSConfigProvider.java:31)
	at com.falsepattern.zigbrains.zig.lsp.ZLSStreamConnectionProvider.getCommand(ZLSStreamConnectionProvider.java:47)
	at com.falsepattern.zigbrains.zig.lsp.ZLSStreamConnectionProvider.<init>(ZLSStreamConnectionProvider.java:26)
	at com.falsepattern.zigbrains.zig.lsp.ZLSLanguageServerFactory.createConnectionProvider(ZLSLanguageServerFactory.java:14)
	at com.redhat.devtools.lsp4ij.server.definition.extension.ExtensionLanguageServerDefinition.createConnectionProvider(ExtensionLanguageServerDefinition.java:54)
	... 8 more
java.lang.RuntimeException: Exception occurred while creating an instance of the stream connection provider
	at com.redhat.devtools.lsp4ij.server.definition.extension.ExtensionLanguageServerDefinition.createConnectionProvider(ExtensionLanguageServerDefinition.java:56)
	at com.redhat.devtools.lsp4ij.LanguageServerWrapper.lambda$start$0(LanguageServerWrapper.java:223)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: java.lang.NullPointerException
	at java.base/java.util.Objects.requireNonNull(Objects.java:209)
	at java.base/sun.nio.fs.UnixFileSystem.getPath(UnixFileSystem.java:263)
	at java.base/java.nio.file.Path.of(Path.java:147)
	at com.falsepattern.zigbrains.project.toolchain.ToolchainZLSConfigProvider.lambda$getEnvironment$0(ToolchainZLSConfigProvider.java:38)
	at java.base/java.util.Optional.ifPresent(Optional.java:178)
	at com.falsepattern.zigbrains.project.toolchain.ToolchainZLSConfigProvider.getEnvironment(ToolchainZLSConfigProvider.java:38)
	at com.falsepattern.zigbrains.zig.environment.ZLSConfigProvider.findEnvironment(ZLSConfigProvider.java:31)
	at com.falsepattern.zigbrains.zig.lsp.ZLSStreamConnectionProvider.getCommand(ZLSStreamConnectionProvider.java:47)
	at com.falsepattern.zigbrains.zig.lsp.ZLSStreamConnectionProvider.<init>(ZLSStreamConnectionProvider.java:26)
	at com.falsepattern.zigbrains.zig.lsp.ZLSLanguageServerFactory.createConnectionProvider(ZLSLanguageServerFactory.java:14)
	at com.redhat.devtools.lsp4ij.server.definition.extension.ExtensionLanguageServerDefinition.createConnectionProvider(ExtensionLanguageServerDefinition.java:54)
	... 8 more

```

**Comment by FalsePattern** [comment]
Looks like your standard library location is a relative path, it needs to be an absolute path
![image](https://github.com/user-attachments/assets/d55f9627-191a-49af-8fef-7ee176811ff3)

**Comment by BezBartek** [comment]
I updated plugin toady, to **16.1.0-241** and it started working without any change on my side

**Comment by FalsePattern** [comment]
I did notice one odd behaviour on my system too, sometimes ZLS wouldn't start no matter what, but restarting the IDE would suddenly fix it, so it might have been that.

I'll keep this issue open for now, please post a message if it ever happens again.

**Comment by FalsePattern** [comment]
The relative path thing should be fixed in 17.1.0

**Comment by FalsePattern** [comment]
@BezBartek Does this still happen?
